crates/conversions/src/serde/serialize/serialize_impl.rs

```diff
@@ -27,14 +29,28 @@ impl CairoSerialize for CallEntryPoint {
 
 impl CairoSerialize for ContractErrorData {
     fn serialize(&self, output: &mut BufferWriter) {
-        ByteArray::from(self.revert_error.as_str()).serialize(output);
+        self.revert_error.serialize(output);
+    }
+}
+
+impl CairoSerialize for ContractExecutionError {
+    fn serialize(&self, output: &mut BufferWriter) {
+        match &self {
+            ContractExecutionError::Nested(inner) => {
+                inner.class_hash.serialize(output);
+                inner.contract_address.serialize(output);
+                inner.selector.serialize(output);
+                inner.error.serialize(output);
+            }
+            ContractExecutionError::Message(msg) => ByteArray::from(msg.as_str()).serialize(output),
+        }
```

Comment:
Do we actually test these anywhere? I think it would be worth doing so.

Reply:
I made output assertion in `test_wrong_function_name` and `test_wrong_calldata` more detailed, so now it checks the serialization, take a look.

Reply:
Okay, ideally we'd have some tests here as well but okay.